spring/src/main/java/org/axonframework/spring/config/SpringAxonAutoConfigurer.java

```diff
@@ -244,8 +245,11 @@ private void registerCorrelationDataProviders(Configurer configurer) {
     }
 
     private void registerEventUpcasters(Configurer configurer) {
-        Arrays.stream(beanFactory.getBeanNamesForType(EventUpcaster.class))
-              .forEach(name -> configurer.registerEventUpcaster(c -> getBean(name, c)));
+        beanFactory.getBeansOfType(EventUpcaster.class).values()
+                   .stream()
+                   .sorted(AnnotationAwareOrderComparator.INSTANCE)
+                   .collect(Collectors.toList())
```

Comment:
No need to collect before using forEach.

Reply:
I assumed the same, but I found some posts that stated `sorted` does not affect the ordering of the stream until a terminating operation.
Hence why I kept `collect` here. Granted, I have not verified this actually... I will still do that. So, thanks for pointing it out!

Reply:
Wait, `forEach` is also one of those of course... I should get more sleep. Thanks again for pointing it out.